zio-http/shared/src/main/scala/zio/http/Body.scala

```diff
@@ -568,27 +570,32 @@ object Body {
     override def asStream(implicit trace: Trace): ZStream[Any, Throwable, Byte] =
       ZStream.unwrap {
         ZIO.blocking {
-          for {
-            r <- ZIO.attempt {
+          ZIO.suspendSucceed {
+            try {
               val fs   = new FileInputStream(file)
               val size = Math.min(chunkSize.toLong, file.length()).toInt
 
-              (fs, size)
-            }
-            (fs, size) = r
-          } yield ZStream
-            .repeatZIOOption[Any, Throwable, Chunk[Byte]] {
```

Comment:
Replacing `ZStream.repeatZIOOption` with `ZStream.repeatZIOChunkOption` allows us to remove the `.flattenChunks`

Anyway, `repeatZIOOption` is using `repeatZIOChunkOption` underneath:
![Uploading image.png…]()